build.sh

```diff
@@ -30,6 +30,7 @@ set -eo pipefail
 
 ROOT=`dirname "$0"`
 ROOT=`cd "$ROOT"; pwd`
+MACHINE_TYPE=$(uname -m)
```

Comment:
some machine may not install uname

Reply:
Maybe we could treat it as a environment dependency, which should be installed.